src/platforms/rp2040/src/lib/networkdriver.c

```diff
@@ -211,10 +215,29 @@ static void send_sntp_sync(struct timeval *tv)
     END_WITH_STACK_HEAP(heap, driver_data->global);
 }
 
+// param should be pointer to malloc'd destination to copy device hostname
+// return ok atom or error as atom
+static term get_default_device_name(char *name, GlobalContext *global)
+{
+    uint8_t mac[6];
+    // Device name is used for AP mode ssid (if undefined), and for the
+    // default dhcp_hostname on both interfaces.  It seems the interface
+    // parameter is ignored and both interfaces have the same MAC address.
+    int err = cyw43_wifi_get_mac(&cyw43_state, CYW43_ITF_STA, mac);
+    if (err) {
+        return globalcontext_make_atom(global, ATOM_STR("\x10", "device_mac_error"));
```

Comment:
In general, shouldn't error be a tuple, such as `{error, Reason}`?
Anyway in this specific case I'd let the caller handle the error.